smdebug/core/hook.py

```diff
@@ -652,15 +652,16 @@ def _save_for_tensor(self, tensor_name, tensor_value, check_before_write=True):
             save_collections_for_tensor = save_collections
 
         self._write_for_tensor(tensor_name, tensor_value, save_collections_for_tensor)
-        for s_col in save_collections_for_tensor:
-            if s_col.name in SM_METRIC_COLLECTIONS:
-                np_val = self._make_numpy_array(tensor_value)
-                # Always log loss to Minerva
-                tensor_val = np.mean(np_val)
-                scalar_obj = ScalarCache(
-                    tensor_name, tensor_val, sm_metric=True, write_tb=False, write_event=False
-                )
-                self.scalar_cache.append(scalar_obj)
+        if "_input" not in tensor_name:
```

Comment:
Let's not hardcode it like this. How about

register_loss fn:
   self.collection_manager.get_collection('losses').add_module_tensors(loss_module) 

# this by default only saves the outputs

Reply:
so inputs are not saved by default for loss?

Reply:
also, this alone wont work because the default regex for loss collection is [Ll]oss, so _input_ tensors are included. then we'll have to change add_module_Tensors() to overwrite the existing regex

Reply:
What if the regex for losses collection is changed to exclude input there? Then we won't need to add the not _input check for all the scalar tensors

Reply:
See how regex for weights excludes the ones which have both weights and gradients in them
as gradient tensor names are of the format

gradient/weight/x

Reply:
we'll have to do this in register default collections then. any regex mentioned after this initial register_default_collections call, appends regex to the existing.
 if the user explicitly wants input tensors to be included, then call add_module_tensors() later

Reply:
No, change it here itself https://github.com/awslabs/sagemaker-debugger/blob/e26ef31c29f8dce2768ba09a8a0cf843a9a77c5a/smdebug/pytorch/collection.py#L42

Reply:
Never mind, that's what you referred to as well. Yeah

Reply:
Isn't the input a non-scalar? How is it going to Minerva?